### PR TITLE
Fix domain parsing, and refactor dockerfiles

### PR DESCRIPTION
**Pull request by taha-au**
Also, the test quickstart workflow is being refactored in this PR for readability reasons.

Note for future: the quickstart workflow test doesn't actually test the quickstart script. it simply tests the fetch. we should update it to do more than just that

### CODE REVIEW

File: scripts/quickstart.sh [hunk 1]
```diff
@@ -101,7 +101,7 @@ conf() {
     # Extract SUPERBLOCKS_AGENT_CUSTOM_DOMAIN from SUPERBLOCKS_AGENT_HOST_URL
     if [ "$1" = "SUPERBLOCKS_AGENT_HOST_URL" ]; then
         if [[ $2 == http* ]]; then
-            domain=$(awk -F/ '{print $3}' <<<"$2")
+            domain=$(echo "$2" | awk -F/ '{print $3}' -)
```
Comment (taha-au) [inline scripts/quickstart.sh:104]: `<<<` is not posix compliant

File: scripts/quickstart.sh [hunk 2]
```diff
@@ -101,7 +101,7 @@ conf() {
     # Extract SUPERBLOCKS_AGENT_CUSTOM_DOMAIN from SUPERBLOCKS_AGENT_HOST_URL
     if [ "$1" = "SUPERBLOCKS_AGENT_HOST_URL" ]; then
         if [[ $2 == http* ]]; then
```
Comment (bruce-y) [inline scripts/quickstart.sh:103]: speaking of posix compliance, `[[ ]]` is not posix compliant either. If we don't expect users to run using bash, we should fix this everywhere

Comment (bruce-y) [reply scripts/quickstart.sh:103]: looks like it's just this instance

Comment (taha-au) [reply scripts/quickstart.sh:103]: nice catch, fixing. and yeah, the shebang is set to `sh`, which is what im going by

**Review by bruce-y — APPROVED** [review]
👌

**Pull request merged**